modules/services/screen-locker.nix

```diff
@@ -62,7 +62,7 @@ in {
           ${pkgs.xautolock}/bin/xautolock \
           -detectsleep \
           -time ${toString cfg.inactiveInterval} \
-          -locker '${pkgs.systemd}/bin/loginctl lock-session $XDG_SESSION_ID' \
+          -locker '${pkgs.systemd}/bin/loginctl lock-session $XDG_SESSION_ID' ${lib.optionalString (cfg.xautolockExtraOptions != []) "\\"}
```

Comment:
Why aren't we just appending conditionally to the end?

e.g.:

```nix
{
  foo = ''
    … -locker '${pkgs.systemd}/bin/loginctl lock-session $XDG_SESSION_ID' ''
  + lib.optionalString (cfg.xautolockExtraOptions != []) (concatStringsSep " " cfg.xautolockExtraOptions);
}
```

Reply:
I guess it's to make the unit file itself a bit more readable.

Reply:
Hmm, I would suggest to simply change to

```
ExecStart = concatStringsSep " " [
  "${pkgs.xautolock}/bin/xautolock"
  "-detectsleep"
  "-time ${toString cfg.inactiveInterval}"
  "-locker '${pkgs.systemd}/bin/loginctl lock-session $XDG_SESSION_ID'"
]
++ cfg.xautolockExtraOptions;
```

IMHO it's not terribly important to keep the generated file super clean. Better keep it robust and simple 🙂

Reply:
I couldn't agree more. Updated :-)